packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -261,7 +273,7 @@ class DatePicker extends React.Component {
 
     const portalPicker = (
       <ReactDatePicker
-        {...customProps}
+        {...attributes}
```

Comment:
I'm not sure that this is going to work as expected because these attributes are spread onto ReactDatePicker and I don't think it takes custom props. The first class `required` prop should be added to the inputAttributes prop in DateInput. Adding `aria-required` and setting the `required` prop to Input would need to be done in [DateInput](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-picker/src/DateInput.jsx#L161-L170).